### PR TITLE
[RuntimeAsync] Add x86 support

### PR DESCRIPTION
**Pull request by jakobbotsch**
This PR makes all our async2 tests pass on win-x86. The main thing needed is that the JIT32 GC encoder has a hard limit on 4 epilogs, so the JIT needs to share suspension epilogs, and also needs to ensure we have at most 3 other normal epilogs when we get to the async2 transformation.

Contributes to #2764

### CODE REVIEW

File: src/coreclr/jit/async.h [hunk 1]
```diff
@@ -32,6 +32,7 @@ class Async2Transformation
     unsigned                      m_exceptionVar            = BAD_VAR_NUM;
     BasicBlock*                   m_lastSuspensionBB        = nullptr;
     BasicBlock*                   m_lastResumptionBB        = nullptr;
+    BasicBlock* m_sharedReturnBB                            = nullptr;
```
Comment (jakobbotsch) [inline src/coreclr/jit/async.h:35]: Formatting is a bit off here, but the JIT formatter makes a bunch of changes in various code, so I want to submit those formatting changes in a separate PR.

**Comment by jakobbotsch** [comment]
PTAL @VSadov

File: src/coreclr/jit/fgdiagnostic.cpp [hunk 2]
```diff
@@ -2726,6 +2726,9 @@ bool BBPredsChecker::CheckEhTryDsc(BasicBlock* block, BasicBlock* blockPred, EHb
         return true;
     }
 
+    // Async resumptions are allowed to jump into try blocks at any point. They
+    // are introduced late enough that the invariant of single entry is no
+    // longer necessary.
```
Comment (VSadov) [inline src/coreclr/jit/fgdiagnostic.cpp:2731]: Roslyn rewrites awaits in `try` blocks as a switch at the single entry that dispatches to appropriate resume location. Nested cases get dispatched multiple times. It is nice that runtime async can resume directly into `try` blocks, so we can leave awaits in trys as-is.

Awaits in handlers might be harder to allow, I guess.

**Review by VSadov — APPROVED** [review]
LGTM. Thanks!!

File: src/coreclr/jit/fgdiagnostic.cpp (hunk 2 above)
Comment (jakobbotsch) [reply src/coreclr/jit/fgdiagnostic.cpp:2731]: > Roslyn rewrites awaits in `try` blocks as a switch at the single entry that dispatches to appropriate resume location. Nested cases get dispatched multiple times. It is nice that runtime async can resume directly into `try` blocks, so we can leave awaits in trys as-is.

Yeah, this definitely simplifies things. For OSR we have to do the similar transformation as Roslyn, creating the "step" blocks at the beginning of try's.

> Awaits in handlers might be harder to allow, I guess.

Indeed, seems like it would be very hard, so I'm glad we disallow that.
